pkg/devfile/adapters/common/command_test.go

```diff
@@ -10,215 +10,126 @@ import (
-					Workdir:   &workDir[0],
-					Type:      &validCommandType,
+					CommandLine: emptyString,
+					Component:   components[0],
+					WorkingDir:  workDir[0],
+					Group:       &versionsCommon.Group{Kind: initGroup},
 				},
 			},
-			isCommandRequired: []bool{false},
-			wantErr:           true,
+			requestedType: []common.DevfileCommandGroupType{initGroup},
+			wantErr:       true,
 		},
 		{
-			name:              "Case 10: Invalid devfile with empty devbuild command",
-			requestedCommands: []string{"devbuild"},
-			commandActions: []common.DevfileCommandAction{
+			name: "Case 6: Valid devfile with empty workdir",
+			execCommands: []common.Exec{
 				{
-					Command:   &emptyString,
-					Component: &components[0],
-					Workdir:   &workDir[0],
-					Type:      &validCommandType,
+					CommandLine: commands[0],
+					Component:   components[0],
+					Group:       &versionsCommon.Group{Kind: runGroup},
 				},
 			},
-			isCommandRequired: []bool{false},
-			wantErr:           true,
+			requestedType: []common.DevfileCommandGroupType{runGroup},
+			wantErr:       false,
 		},
 		{
-			name:              "Case 11: Valid devfile with empty workdir",
-			requestedCommands: []string{"devrun"},
-			commandActions: []common.DevfileCommandAction{
+			name: "Case 7: Invalid command referencing an absent component",
+			execCommands: []common.Exec{
 				{
-					Command:   &commands[0],
-					Component: &components[0],
-					Type:      &validCommandType,
+					CommandLine: commands[0],
+					Component:   invalidComponent,
+					Group:       &versionsCommon.Group{Kind: runGroup},
 				},
 			},
-			isCommandRequired: []bool{true},
-			wantErr:           false,
-		},
-		{
-			name:              "Case 12: Invalid command referencing an absent component",
-			requestedCommands: []string{"devrun"},
-			commandActions: []common.DevfileCommandAction{
-				{
-					Command:   &commands[0],
-					Component: &invalidComponent,
-					Type:      &validCommandType,
-				},
-			},
-			isCommandRequired: []bool{true},
-			wantErr:           true,
+			requestedType: []common.DevfileCommandGroupType{runGroup},
+			wantErr:       true,
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
+			components := []common.DevfileComponent{testingutil.GetFakeComponent(tt.execCommands[0].Component)}
+			if tt.execCommands[0].Component == invalidComponent {
+				components = []common.DevfileComponent{testingutil.GetFakeComponent("randomComponent")}
+			}
 			devObj := devfileParser.DevfileObj{
 				Data: testingutil.TestDevfileData{
-					CommandActions: tt.commandActions,
-					ComponentType:  common.DevfileComponentTypeDockerimage,
+					ExecCommands: tt.execCommands,
+					Components:   components,
 				},
 			}
 
-			for i, commandName := range tt.requestedCommands {
-				command, err := getCommand(devObj.Data, commandName, tt.isCommandRequired[i])
+			for _, gtype := range tt.requestedType {
+				_, err := getCommand(devObj.Data, "", gtype)
```

Comment:
the command name shouldn't be empty

Reply:
this scenario is covered for all three commands, check tests for `TestGetBuildCommand` `https://github.com/openshift/odo/pull/3220/files#diff-0e2dae7fac15ec15148977126c96a904R346`

Reply:
OK